pytorch_lightning/plugins/environments/cluster_environment.py

```diff
@@ -11,24 +11,33 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from abc import ABC, abstractmethod
+from typing import Optional
 
 
-class ClusterEnvironment:
+class ClusterEnvironment(ABC):
+    """ Specification of a cluster environment. """
 
-    def __init__(self):
-        self._world_size = None
+    @abstractmethod
+    def spawns_children(self) -> bool:
```

Comment:
Should all of those be functions or properties instead?

Reply:
could be, not sure, added it to #6303
for this PR I just followed the design that was already there.